router/router_test.go

```diff
@@ -313,3 +391,58 @@ func wsServer(t *testing.T) {
 	}
 	log.Println("Out of server")
 }
+
+func TestUnitGetRouterFromConfigService(t *testing.T) {
+	t.Log("Testing: Retrieving router port from the config service, after router service startup")
+	expectedValue := "5002"
+	response, err := http.DefaultClient.Get(ConfigServiceUrl +"/config-service/variable/router_http_port")
+	if err != nil {
+		t.Errorf("Error getting router_http_port from config service: %v", err)
+		return
+	}
+	defer response.Body.Close()
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		t.Errorf("Error reading response data")
+		return
+	}
+	if response.StatusCode != 200 {
+		t.Errorf("Error response from config service: StatusCode: %v, Body: %v", response.StatusCode, string(body[:]))
+		return
+	}
+	if string(body[:]) != expectedValue {
+		t.Errorf("Expected value in response to be %v, instead %v", expectedValue, string(body[:]))
+		return
+	}
+
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Integration Tests
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+func TestIntegrationRetrieveRouterPortFromConfigServiceAfterRouterStartup(t *testing.T) {
```

Comment:
Well, before the test itself runs, this gets run: TestMain(). This calls a startup, and teardown before and after every test that runs. The router services gets started on startup, and on startup the router service calls addPortToConfigService(). I can put a comment in the code, saying this above the test to be more comprehensive, but i dont think its necessary to explicitly call that function in the test